### PR TITLE
US-12634 Language toggle implementation

### PR DESCRIPTION
**Pull request by edwinjebaraj**
1. **LanguageToggle\index.tsx**   - Declared new Pcore variable to store the toggled language value and locale reference value
2. **Assignment.tsx** -  Subscribed language toggle value and updated the title based on language .
3. **View.tsx** -  loaded LocaleResources based on language toggle
4. **ClaimsList\index.tsx** - Added language toggle dependency variable to reload the claimsList data. 
5. **ChildBenefitsClaim\index.tsx** - Subscribed language toggle value into state variable and pass it through ClaimsList component as props.

### CODE REVIEW

File: src/components/AppComponents/LanguageToggle/index.tsx [hunk 1]
```diff
@@ -23,6 +23,9 @@ const LanguageToggle = props => {
         '@BASECLASS!DATAPAGE!D_LISTREFERENCEDATABYTYPE',
         'HMRC-CHB-WORK-CLAIM!CASE!CLAIM'
       ]);
+
+      PCore.getPubSubUtils().publish('languageToggleTriggered', {language: lang});
+      PCore.getPubSubUtils().publish('languageToggleLocale', []);
```
Comment (gautamkumar-hmrc) [inline src/components/AppComponents/LanguageToggle/index.tsx:28]: why we are publishing multiple events, we can listen the same event on multiple location.

File: src/components/override-sdk/infra/Assignment/Assignment.tsx [hunk 2]
```diff
@@ -68,6 +68,7 @@ export default function Assignment(props) {
   const [errorMessages, setErrorMessages] = useState<Array<OrderedErrorMessage>>([]);
   const [serviceShutteredStatus, setServiceShutteredStatus] = useState(serviceShuttered);
   const [header, setHeader] = useState('');
+  const [selectedLang, setSelectedLang] = useState('');
```
Comment (gautamkumar-hmrc) [inline src/components/override-sdk/infra/Assignment/Assignment.tsx:71]: default value should be English

File: src/components/AppComponents/LanguageToggle/index.tsx (hunk 1 above)
Comment (edwinjebaraj) [reply src/components/AppComponents/LanguageToggle/index.tsx:28]: I have added PCore.getPubSubUtils().publish('languageToggleLocale', []); event to check and store localeReference values.

File: src/components/override-sdk/infra/Assignment/Assignment.tsx (hunk 2 above)
Comment (edwinjebaraj) [reply src/components/override-sdk/infra/Assignment/Assignment.tsx:71]: I will update this.

File: src/components/override-sdk/infra/View/View.tsx [hunk 3]
```diff
@@ -26,10 +26,19 @@ const NO_HEADER_TEMPLATES = [
...
 ];
 
+declare const PCore: any;
 export default function View(props) {
   const { children, template, getPConnect, mode, visibility, name: pageName } = props;
   let { label, showLabel = false } = props;
 
+  // call the locale reference api when we toggle the languge from English to Welsh
+  PCore.getPubSubUtils().subscribe('languageToggleTriggered', locale => {
+    if(!locale.localeRef.includes(props.localeReference)) {
```
Comment (gautamkumar-hmrc) [inline src/components/override-sdk/infra/View/View.tsx:36]: While publishing event you have passed localeRef=[], so i don't see any point of using if block.

**Review by gautamkumar-hmrc — APPROVED** [review]

**Review by gautamkumar-hmrc — APPROVED** [review]

**Review by AkhilaHMRC — APPROVED** [review]

**Pull request merged**